doc/README.md

```diff
@@ -0,0 +1,136 @@
+
+This proposal outlines an approach for adding GRPC client support to Tailcall to allow calling GRPC services from Tailcall resolvers.
+
+
+## Design Goals
+
+The primary design goals are:
+
+- Call GRPC services from Tailcall resolvers
+- Generate GRPC client stubs from GraphQL schema
+- Map between GraphQL and GRPC requests/responses
+- No changes needed to Tailcall server
+
+
+## Proposed Solution
+
+The proposed solution is to:
+
+Add a grpc section to the Tailcall config file, similar to http to configure GRPC endpoints:
+
+```bash
+grpc
+  endpoints = [
+    {port = 50051, proto="user.proto"},
+    {port = 50052, proto="product.proto"}
+  ]
+```
+
+
+Add a @grpc directive to mark GraphQL types as GRPC messages:
+
+```bash
+@grpc
+type User {
+  id: ID! 
+  name: String!
+}
+```
+- Generate GRPC proto files from GraphQL schemas with @grpc types.
+- Use a GRPC client library like tonic to generate GRPC client stubs.
+- Call the GRPC client stub from Tailcall resolvers, handling request/response mapping.
+
+
+## Detailed Implementation using Tonic
+
+Tonic is a gRPC over HTTP/2 implementation focused on high performance, interoperability, and flexibility. It's written in Rust, making it an excellent choice for implementing the gRPC feature in Tailcall. Here's a detailed section on how we can use Tonic:
+
+1. Add Tonic and Prost to Your Dependencies: 
+
+Tonic is the gRPC client and server implementation, and Prost is a Protocol Buffers implementation in Rust. Add them to your `Cargo.toml` file:
+
+```bash
+[dependencies]
+tonic = "0.5"
+prost = "0.9"
+```
+
+2. Generate gRPC Client Stubs
+
+- Add @grpc directive to GraphQL schema to specify GRPC service and method for each field.
+- Run a custom tool on startup to parse schema and generate .proto files for @grpc fields.
+- The tool extracts the service, method, request and response types.
+- It generates .proto files defining the RPC service and messages.
+
+```bash
+graphql
+type User @grpc(service: "user.UserService") {
+  name: String @grpc(method: "GetUserName")
+  age: Int @grpc(method: "GetUserAge") 
+}
+```
+
+Parse the schema and generate the corresponding .proto files. Tonic can then generate Rust code from the .proto files.
+
+3. Handle Schema Conversion
+
+- Create a new grpc module to handle gRPC code generation. This module would contain the logic to parse  the schema and generate .proto files.
+- In the schema generation code, after creating the JSON schema, call the new grpc module to handle gRPC schema generation.
+- The grpc module would parse the schema AST and look for types annotated with @grpc directives. It would extract the service name, method name, request and response types for each gRPC endpoint.
+- Using this extracted information, the grpc module can generate a .proto file defining the gRPC service, messages, and RPC methods.
+- The grpc module should write the generated .proto files to disk, in a grpc sub-directory alongside the JSON schemas.
```

Comment:
I don't think we should write the files to disk. It should stay in memory like how we keep `JsonSchema` when we want to perform response validation